app/src/main/java/com/maxwai/nclientv3/MainActivity.java

```diff
@@ -717,6 +697,29 @@ public void onNothingSelected(AdapterView<?> parent) {
         builder.show();
     }
 
+    private void updateLanguageAndIcon(MenuItem item) {
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.select_dialog_singlechoice);
+        MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(this);
+
+        for (OnlyLanguage lang : OnlyLanguage.values()) {
+            adapter.add(getString(lang.getNameId()));
+        }
+        selectedLanguage = OnlyLanguage.findFromLanguages(Global.getOnlyLanguage());
+        builder.setIcon(R.drawable.ic_world)
+            .setTitle(R.string.change_language)
+            .setSingleChoiceItems(adapter, selectedLanguage.ordinal(),
+                (dialog, which) -> selectedLanguage = OnlyLanguage.values()[which])
+            .setPositiveButton(R.string.ok, (dialog, which) -> {
+                Global.updateOnlyLanguage(MainActivity.this, Language.valueOf(selectedLanguage.name()));
+                changeLanguageTimeHandler.removeCallbacks(changeLanguageRunnable);
```

Comment:
Add back the comment about the delay
```suggestion
                // wait 250ms to reduce the requests
                changeLanguageTimeHandler.removeCallbacks(changeLanguageRunnable);
```